pymatgen/io/vasp/outputs.py

```diff
@@ -721,25 +721,31 @@ def get_computed_entry(self, inc_structure=True, parameters=None,
             return ComputedStructureEntry(self.final_structure, self.final_energy, parameters=params, data=data)
         return ComputedEntry(self.final_structure.composition, self.final_energy, parameters=params, data=data)
 
-    def get_band_structure(self, kpoints_filename=None, efermi=None,
-                           line_mode=False, force_hybrid_mode=False):
+    def get_band_structure(self,
+                           kpoints_filename: Optional[str] = None,
+                           efermi: Union[str, float, None] = "smart",
```

Comment:
I'm not sure about modifying the Fermi level by default?

Seems like it could lead to a lot of confusion for people who aren't aware of this option.

Reply:
I agree, thanks @utf.

Reply:
> The Fermi energy associated with the bandstructure, in eV. By default,
                 uses the 'smart_efermi' attribute of the Vasprun, which in some cases
                 may differ from (but be more accurate than) the value in vasprun.xml.

This docstring should also be reworded to state explicitly what this does (ie remove the reference to "smart"). I don't think we should making claims as to the accuracy though it *can* be more accurate.

The type hint could be `Optional[Union[float, str]]` too.